server/src/main/java/org/opensearch/search/aggregations/bucket/terms/InternalSignificantTerms.java

```diff
@@ -265,13 +289,17 @@ public InternalAggregation reduce(List<InternalAggregation> aggregations, Reduce
             }
         }
         SignificanceHeuristic heuristic = getSignificanceHeuristic().rewrite(reduceContext);
-        final int size = reduceContext.isFinalReduce() == false ? buckets.size() : Math.min(requiredSize, buckets.size());
+        final int size = (reduceContext.isFinalReduce() || reduceContext.isSliceLevel())
+            ? Math.min(requiredSizeLocal, buckets.size())
+            : buckets.size();
         BucketSignificancePriorityQueue<B> ordered = new BucketSignificancePriorityQueue<>(size);
         for (Map.Entry<String, List<B>> entry : buckets.entrySet()) {
             List<B> sameTermBuckets = entry.getValue();
             final B b = reduceBucket(sameTermBuckets, reduceContext);
             b.updateScore(heuristic);
-            if (((b.score > 0) && (b.subsetDf >= minDocCount)) || reduceContext.isFinalReduce() == false) {
+            if (!(reduceContext.isFinalReduce() || reduceContext.isSliceLevel()) // Don't apply thresholds for partial reduce
```

Comment:
The condition is unreadable to be fair, could you please introduce local variables instead? That would help to understand why the specific path is taken.

Reply:
Addressed in the latest commit